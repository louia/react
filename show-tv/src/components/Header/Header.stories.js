import React from 'react';
import Header from './index';
import '../../index.css'

export default {
    title: 'components/Header',
    component: Header,
}

export const default_value = () => <Header />;
export const initialized = () => <Header  title="Bienvenue" subtitle="Sous titre" isMiniature={false}/>;
export const mini = () => <Header title="Bienvenue" subtitle="Sous titre" isMiniature={true}/>;
