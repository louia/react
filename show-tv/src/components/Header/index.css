header{
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content:center;
    align-items: center;
    height: 8em;
    border: black solid 1px;
    border-radius: 15px;
  }
  
  .item-header:first-child{
    font-size: 1.5em;
    font-weight: bold;
  }

  .item-header:last-child{
    font-size: 1.1em;
    font-weight: bold;
  }

  p{
    margin:0 !important;
  }