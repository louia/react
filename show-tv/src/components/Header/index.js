import React from 'react';
import PropTypes from 'prop-types';

export class Header extends React.Component {
    render() {
        if (this.props.isMiniature == true) {
            return (
                <header className="mini">
                    <div className="item-header">{this.props.title}</div>
                    <div className="item-header">{this.props.subtitle}</div>
                </header>
            );
        }
        return (
            <header>
                <div className="item-header">{this.props.title}</div>
                <div className="item-header">{this.props.subtitle}</div>
            </header>
        );
    }
}


Header.propTypes = {
    isMiniature: PropTypes.bool,
    title: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.instanceOf(Element)
    ]),

    subtitle: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.instanceOf(Element)
    ]),
};


export default Header;
