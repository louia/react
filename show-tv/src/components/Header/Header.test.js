import { render, fireEvent } from '@testing-library/react';
import { default_value, initialized, mini } from './Header.stories';
import initStoryshots from '@storybook/addon-storyshots';
initStoryshots({ /* configuration options */ });


describe('Header test', () => {

    test('should increment when clicked', () => {
        expect(true).toBeTruthy();
    });
    // test('Header has container className', () => {
    //     const header = render(mini);
    //     console.log(header);
        
    //     expect(header.classList.contains('item')).toBe(true)
    // });
    // test('Header child has item className', () => {
    //     const header = render(mini);
    //     expect(header.child[0].classList.contains('item')).toBe(true)
    //     expect(header.child[1].classList.contains('item')).toBe(true)
    // });
});
