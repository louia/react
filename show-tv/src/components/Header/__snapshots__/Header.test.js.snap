// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components/Header Default Value 1`] = `
<header>
  <div
    className="item-header"
  />
  <div
    className="item-header"
  />
</header>
`;

exports[`Storyshots components/Header Initialized 1`] = `
<header>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;

exports[`Storyshots components/Header Mini 1`] = `
<header
  className="mini"
>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;
